synapse/storage/databases/main/state.py

```diff
@@ -133,6 +145,36 @@ def get_room_version_id_txn(self, txn: LoggingTransaction, room_id: str) -> str:
 
         return room_version
 
+    async def get_metadata_for_events(
+        self, event_ids: Collection[str]
+    ) -> Dict[str, EventMetadata]:
+        """Get some metadata (room_id, type, state_key) for the given events."""
```

Comment:
this is all a bit woolly. What counts as metadata? Why shouldn't I just call `get_events`? 

Can you add some guidelines for future developers for what would and would not be sensible to return here, and also when it is appropriate to call this method instead of `get_events` ?